From f4754af39260d0a9f86dd82251cc4004bc17d76d Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 29 Oct 2008 10:28:53 +0800
Subject: [PATCH 026/259] ENGR00096973 vpu: remove max frame size limit and
 enlarge max filename length

Remove max encoder size limitation from lib.
Enlarge max input/output filename length in test code.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.h  | 3 ++-
 vpu/vpu_util.c | 5 +----
 vpu/vpu_util.h | 3 ---
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index e633433..7e46509 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -457,11 +457,12 @@ typedef struct vpu_versioninfo {
 
 /*
  * Revision History:
+ * v4.3.2 [2008.10.28] support loopback in MX51
  * v4.2.2 [2008.09.03] support encoder in MX51
  * v4.1.2 [2008.08.22] update MX37 VPU firmware to V1.0.5
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(4, 1, 2)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(4, 3, 2)
 
 extern unsigned int system_rev;
 
diff --git a/vpu/vpu_util.c b/vpu/vpu_util.c
index ebdecb6..88bf3e7 100644
--- a/vpu/vpu_util.c
+++ b/vpu/vpu_util.c
@@ -384,10 +384,7 @@ RetCode CheckEncOpenParam(EncOpenParam * pop)
 		}
 	}
 
-	if (picWidth < 32 || picWidth > MAX_ENC_PIC_WIDTH) {
-		return RETCODE_INVALID_PARAM;
-	}
-	if (picHeight < 16 || picHeight > MAX_ENC_PIC_HEIGHT) {
+	if (picWidth < 32 || picHeight < 16) {
 		return RETCODE_INVALID_PARAM;
 	}
 
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index dfecef5..abe16b1 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -78,9 +78,6 @@ enum {
 	FIRMWARE_GET = 0xf
 };
 
-#define MAX_ENC_PIC_WIDTH		720
-#define MAX_ENC_PIC_HEIGHT		576
-
 void BitIssueCommand(int instIdx, int cdcMode, int cmd);
 
 RetCode LoadBitCodeTable(Uint16 * pBitCode, int *size);
-- 
1.8.0

